src/process/base/ZLibBasedFASTQProcessorBaseClass.h

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2019 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#ifndef FASTQINDEX_ZLIBHELPER_H
+#define FASTQINDEX_ZLIBHELPER_H
+
+#include "common/CommonStructsAndConstants.h"
+#include "common/ErrorAccumulator.h"
+#include "process/extract/IndexReader.h"
+#include "process/io/Source.h"
+#include "process/io/Sink.h"
+#include "process/io/PathSource.h"
+#include <cstdio>
+#include <experimental/filesystem>
+#include <zlib.h>
+
+using std::experimental::filesystem::path;
+using std::stringstream;
+
+/**
+ * Helper class for zlib stuff.
+ */
+class ZLibBasedFASTQProcessorBaseClass : public ErrorAccumulator {
```

Comment:
Given that the tool does not only work with FASTQs, did you consider removing all "FASTQ" references from the code?